In [ ]:
from google.colab import files
uploaded = files.upload()
for fn in uploaded.keys():
  print('User uploaded file "{name}" with length {length} bytes'.format(name=fn,length=len(uploaded[fn])))

Saving Restaurant_Reviews.tsv to Restaurant_Reviews.tsv
User uploaded file "Restaurant_Reviews.tsv" with length 61332 bytes


In [ ]:
import numpy as np
import matplotlib.pyplot as plt 
import pandas as pd
dataset = pd.read_csv('Restaurant_Reviews.tsv', delimiter = '\t')
dataset.head()

Review  Liked
0                           Wow... Loved this place.      1
1                                 Crust is not good.      0
2          Not tasty and the texture was just nasty.      0
3  Stopped by during the late May bank holiday of...      1
4  The selection on the menu was great and so wer...      1

In [ ]:
len(dataset)

1000

In [ ]:
dataset.isnull().sum()

Review    0
Liked     0
dtype: int64

In [ ]:
#cleaning the text
import re
review=re.sub('[^a-zA-Z]',' ',dataset['Review'][0])

In [ ]:
review

'Wow    Loved this place '

In [ ]:
#cleaning the text and the text in lower case
import re
review=re.sub('[^a-zA-Z]',' ',dataset['Review'][0])
review=review.lower()

In [ ]:
review


'wow    loved this place '

In [ ]:
#cleaning the text and the text in lower case using split()
import re
review=re.sub('[^a-zA-Z]',' ',dataset['Review'][0])
review=review.lower()
review=review.split()

In [ ]:
review

['wow', 'loved', 'this', 'place']

In [ ]:
import re
import nltk
nltk.download('stopwords')
from nltk.stem.porter import PorterStemmer
from nltk.corpus import stopwords
review=re.sub('[^a-zA-Z]',' ',dataset['Review'][0])
review=review.lower()
review=review.split()
ps=PorterStemmer()
review=[ps.stem(word)for word in review if not word in set(stopwords.words('english'))]
review=' '.join(review)

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


In [ ]:
review

'wow love place'

In [ ]:
import re
import nltk
nltk.download('stopwords')
from nltk.stem.porter import PorterStemmer
from nltk.corpus import stopwords
corpus=[]
for i in range(0,1000):
    review=re.sub('[^a-zA-Z]',' ',dataset['Review'][i])
    review=review.lower()
    review=review.split()
    ps=PorterStemmer()
    review=[ps.stem(word)for word in review if not word in set(stopwords.words('english'))]
    review=' '.join(review)
    corpus.append(review)

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


In [ ]:
#creating the word models
from sklearn.feature_extraction.text import CountVectorizer
cv = CountVectorizer(max_features = 1500)
X = cv.fit_transform(corpus).toarray()
y = dataset.iloc[:,1].values

In [ ]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train,y_test = train_test_split(X, y, test_size = 0.20, random_state = 0)

In [ ]:
#SVC
from sklearn.svm import SVC
classifier = SVC()
classifier.fit(X_train,y_train)

SVC()

In [ ]:
y_pred = classifier.predict(X_test)

In [ ]:
from sklearn import metrics
print(metrics.classification_report(y_test,y_pred))

              precision    recall  f1-score   support

           0       0.66      0.93      0.77        97
           1       0.89      0.55      0.68       103

    accuracy                           0.73       200
   macro avg       0.78      0.74      0.73       200
weighted avg       0.78      0.73      0.73       200



In [ ]:
#making the confusion matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test,y_pred)
print(cm)


[[90  7]
 [46 57]]


In [ ]:
print(metrics.accuracy_score(y_test,y_pred))

0.735


SVC = 0.735
APPROX SVC = 73%

In [ ]:
#GaussianNB 
from sklearn.naive_bayes import GaussianNB
classifier = GaussianNB()
classifier.fit(X_train,y_train)

GaussianNB()

In [ ]:
y_pred = classifier.predict(X_test)

In [ ]:
from sklearn import metrics
print(metrics.classification_report(y_test,y_pred))

              precision    recall  f1-score   support

           0       0.82      0.57      0.67        97
           1       0.68      0.88      0.77       103

    accuracy                           0.73       200
   macro avg       0.75      0.73      0.72       200
weighted avg       0.75      0.73      0.72       200



In [ ]:
#making the confusion matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test,y_pred)
print(cm)

[[55 42]
 [12 91]]


In [ ]:
print(metrics.accuracy_score(y_test,y_pred))

0.73


GaussianNB  = 73%

In [ ]:
#MultinomialNB
from sklearn.naive_bayes import MultinomialNB
classifier = MultinomialNB()
classifier.fit(X_train,y_train)

MultinomialNB()

In [ ]:
y_pred = classifier.predict(X_test)

In [ ]:
from sklearn import metrics
print(metrics.classification_report(y_test,y_pred))

              precision    recall  f1-score   support

           0       0.77      0.74      0.75        97
           1       0.76      0.79      0.78       103

    accuracy                           0.77       200
   macro avg       0.77      0.76      0.76       200
weighted avg       0.77      0.77      0.76       200



In [ ]:
#making the confusion matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test,y_pred)
print(cm)

[[72 25]
 [22 81]]


In [ ]:
print(metrics.accuracy_score(y_test,y_pred))

0.765


MultinomialNB = 76%

In [ ]:
#Neighbours Classifier
from sklearn.neighbors import KNeighborsClassifier
classifier = KNeighborsClassifier(n_neighbors=5)
classifier.fit(X_train,y_train)

KNeighborsClassifier()

In [ ]:
y_pred = classifier.predict(X_test)

In [ ]:
from sklearn import metrics
print(metrics.classification_report(y_test,y_pred))

              precision    recall  f1-score   support

           0       0.56      0.72      0.63        97
           1       0.64      0.46      0.53       103

    accuracy                           0.58       200
   macro avg       0.60      0.59      0.58       200
weighted avg       0.60      0.58      0.58       200



In [ ]:
#making the confusion matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test,y_pred)
print(cm)

[[70 27]
 [56 47]]


In [ ]:
print(metrics.accuracy_score(y_test,y_pred))

0.585


KNeighborsClassifier = 58%

In [ ]:
#LogisticRegression
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression()
classifier.fit(X_train,y_train)

LogisticRegression()

In [ ]:
y_pred = classifier.predict(X_test)

In [ ]:
from sklearn import metrics
print(metrics.classification_report(y_test,y_pred))

              precision    recall  f1-score   support

           0       0.67      0.78      0.72        97
           1       0.76      0.64      0.69       103

    accuracy                           0.71       200
   macro avg       0.72      0.71      0.71       200
weighted avg       0.72      0.71      0.71       200



In [ ]:
#making the confusion matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test,y_pred)
print(cm)

[[76 21]
 [37 66]]


In [ ]:
print(metrics.accuracy_score(y_test,y_pred))

0.71


LogisticRegression = 71%

**SVC = 0.735 APPROX SVC** = 73%
**GaussianNB **= 73%
**MultinomialNB** = 76%
**KNeighborsClassifier **= 58%
**LogisticRegression **= 71% 